apps/site/pages/en/learn/asynchronous-work/discover-promises-in-nodejs.md

```diff
@@ -0,0 +1,317 @@
+
+Promise.any([api1, api2])
+  .then(result => {
+    console.log(result); // 'API 2 success' (since it resolves first)
+  })
+  .catch(error => {
+    console.error('All promises rejected:', error);
+  });
+```
+
+### **`Promise.reject()` and `Promise.resolve()`**:
+
+These methods create a rejected or resolved Promise directly.
+
+```js
+Promise.resolve('Resolved immediately').then(result => {
+  console.log(result); // 'Resolved immediately'
+});
+```
+
+## Error Handling with Promises
+
+Handling errors in Promises ensures your application behaves correctly in case of unexpected situations.
+
+- You can use `.catch()` to handle any errors or rejections that occur during the execution of Promises.
+
+```js
+myPromise
+  .then(result => console.log(result))
+  .catch(error => console.error(error)); // Handles the rejection
+```
+
+- Alternatively, when using `async/await`, you can use a `try/catch` block to catch and handle errors.
+
+```js
+async function performTask() {
+  try {
+    const result = await myPromise;
+    console.log(result);
+  } catch (error) {
+    console.error(error); // Handles any errors
+  } finally {
+    // This code is executed regardless of failure
+    console.log('performTask() completed');
+  }
+}
+
+performTask();
+```
+
+## Scheduling Tasks in the Event Loop
+
+In addition to Promises, Node.js provides several other mechanisms for scheduling tasks in the event loop.
+
+### **`queueMicrotask()`**
+
+`queueMicrotask()` is used to schedule a microtask, which is a lightweight task that runs after the currently executing script but before any other I/O events or timers. Microtasks include tasks like Promise resolutions and other asynchronous operations that are prioritized over regular tasks.
+
+```js
+queueMicrotask(() => {
+  console.log('Microtask is executed');
+});
+
+console.log('Synchronous task is executed');
+```
+
+In the above example, "Microtask is executed" will be logged after "Synchronous task is executed," but before any I/O operations like timers.
+
+### **`process.nextTick()`**
+
+`process.nextTick()` is used to schedule a callback to be executed after the current operation completes, but before any other event loop phases, such as I/O events, timers, or Promises. This is useful for situations where you want to ensure that a callback is executed as soon as possible, but still after the current execution context.
+
+```js
+process.nextTick(() => {
+  console.log('Next tick callback');
+});
+
+console.log('Synchronous task executed');
+```
+
+### **`setImmediate()`**:
+
+`setImmediate()` is used to execute a callback after the current event loop cycle finishes and all I/O events have been processed. This means that `setImmediate()` callbacks run after any I/O callbacks, but before timers and the next tick queue.
+
+```js
+setImmediate(() => {
+  console.log('Immediate callback');
+});
+
+console.log('Synchronous task executed');
+```
+
+### **When to Use Each**:
```

Comment:
It is probably worth stressing that the specific ordering of these is runtime implementation specific... that is, everything about the timing here is very specific to Node.js. While other runtimes might get close, none of the Node.js compatible runtimes adhere to the same ordering on these. For instance, Cloudflare Workers implements `process.nextTick(...)` simply by deferring to `queueMicrotask(...)` (that is, `process.nextTick` is really just an alias for `queueMicrotask`. There are actually quite a few timing differences between the runtimes.